### DISCUSSION TITLE
capstone telegram chatbot error object has no attribute

### DISCUSSION THREAD

**Discussion by budikaryadi** [teamalgoritma/community#379] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/114675693/193436235-30d3ecbb-07a5-455b-b670-a8e92f68aa1f.png)

terdapat error AttributeError: 'int' object has no attribute 'to_dict'.  tipe data unique_campaign sudah disesuaikan menjadi object sama seperti notebook guide

**Comment by yosiaazarya** [comment, marked as answer]
Halo selamat pagi pak Budi, mungkin langkah ini dapat dilakukan saat merubah tipe data dari `campaign_id` -> `df['campaign_id'].astype('str')`

**Reply by budikaryadi** [reply]
![image](https://user-images.githubusercontent.com/114675693/193489034-e9e8f4d5-663c-40f6-becb-ce6049489939.png)

pagi pak, sudah diganti string masih tetap error

**Reply by yosiaazarya** [reply]
Sekarang errornya bukan di bagian baris 91 lagi pak, melainkan bergeser ke baris 105, mungkin boleh dikirimkan pada bagian tersebut?

**Reply by budikaryadi** [reply]
![image](https://user-images.githubusercontent.com/114675693/193491311-52a1b291-2417-46d4-ab0c-19f716a925ba.png)
terlampir ya pak

**Comment by yosiaazarya** [comment]
Mohon dipastikan bahwa kolom `reporting_date` sudah bertipe data waktu, karena `.min()` ataupun `.max()` adalah method yang hanya dapat digunakan untuk tipe data waktu (`datetime64`), terima kasih.

**Reply by budikaryadi** [reply]
![image](https://user-images.githubusercontent.com/114675693/193492660-42afde76-6eaf-47bf-963a-ce62c8bc388f.png)
ini saya coba pd.to_date_time masih error yg sama pak. sebelumnya di read csv sudah parse date. saya coba di notebook guide no issue. tapi ketika coba jalan di .py menjadi error.

**Comment by yosiaazarya** [comment]
pastikan bapak menggunakan accessor `.dt` untuk menjalankan perintah `.strftime()` ya bapak

**Reply by budikaryadi** [reply]
![image](https://user-images.githubusercontent.com/114675693/193494925-019dacb9-8196-4fc0-ad69-4a958bb8e547.png)

sudah ditambah .dt masih error pak

**Comment by yosiaazarya** [comment]
Mohon dipastikan sudah berbentuk `datetime64` dan ditambahkan accessor `.dt` ya pak

**Reply by budikaryadi** [reply]
![image](https://user-images.githubusercontent.com/114675693/193496021-5de9e83f-6b6f-44b1-8bdd-1c32da466f10.png)
![image](https://user-images.githubusercontent.com/114675693/193496047-7a1ccc90-8568-4dce-815b-c48f9fd655af.png)


terlampir pak, ketika import sudah di parse date dan sudah dicoba jg tambah pd.to_datetime dan .dt masih terdapat error sama

**Comment by yosiaazarya** [comment]
Mohon maaf pak, error terjadi karena object `df_campaign` yang bapak buat bukanlah dataframe hasil subsetting namun bapak hanya menggunakan kondisi yaitu `selected_campaign_id`, seharusnya `selected_campaign_id` merupakan kondisi untuk subsetting, sehingga dalam membangun `df_campaign` seharusnya kurang lebih seperti ini :

```
df_campaign = df[selected_campaign_id]
```

**Reply by budikaryadi** [reply]
![image](https://user-images.githubusercontent.com/114675693/193528061-fb1a99ac-2525-4c28-b358-45ee8552ee96.png)
pak kalau error saat deploy gini knp ya? sudah coba tes ganti versi di rutime & requirements masih error

**Reply by IrfanChairurrachman** [reply]
Dear @budikaryadi ,

Pada requirements.txt mohon untuk menghapus `python == 3.10.7` nya ya Pak, karena kemungkinan disebabkan oleh kegagalan install module pada requirements.txt.

Sekian, semoga membantu
